methods.rb

```diff
@@ -0,0 +1,137 @@
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Style/RedundantSelf
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    list = to_a if self.class == Range || Hash
+
+    i = 0
+    while i < element.length
+      yield(list[i])
+      i += 1
+    end
+    self
+  end
```

Comment:
- [ ] Your #my_each when a block is given and when self is an array should behave like #each calls, You

- [ ] Throws an error for:
   - undefined local variable or method `element' for 5..50:Range
   - undefined local variable or method `element' for {:a=>1, :b=>2, :c=>3, :d=>4, :e=>5}:Hash